core/Piranha/Services/PageService.cs

```diff
@@ -707,7 +707,7 @@ private void OnLoad(PageBase model, bool isDraft = false)
                     _factory.Init(model, App.PageTypes.GetById(model.TypeId));
                 }
 
-                App.Hooks.OnLoad<PageBase>(model);
+                App.Hooks.OnLoad(model);
```

Comment:
This will not work with implicit type resolving. The model is of type T which will be a type derived from `PageBase`, but all hooks are registered on `PageBase` which means the Hook Manager will never find any hooks for the model

Reply:
Right, reverted.